### ISSUE TITLE
Tracing Datadog: tag service.name leads to unfortunate behaviour in Datadog

### ISSUE THREAD

**Issue by julientard** [traefik/traefik#8062]
### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?
I activated Datadog tracing 

### What did you expect to see?
I expected to see in the Datadog APM dashboard a consice list of services like :
* traefik
* cms (a app runing in our kube cluster)

### What did you see instead?
I see a huge list of services :
* traefik
* cms
* and many kubeNamespace_cms_randomIdentifier (plz see screeshot attached)
![Screenshot from 2021-04-12 14-17-23](https://user-images.githubusercontent.com/16280863/114572724-09d69b00-9c78-11eb-8e1b-7acf234ea17f.png)

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      2.2.7
Codename:     chevrotin
Go version:   go1.14.6
Built:        2020-07-20T17:06:02Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Traefik is running in a Kubernetes cluster.
Monitoring is done with Datadog, with agents in the cluster.

```toml
[tracing]
  serviceName = "traefik"
  [tracing.datadog]
    prioritySampling = true
    localAgentHostPort = "datadog.datadog.svc:8126"
```

### My understanding of the issue

The Datadog APM dashboard is not really useable because of these random services.
I don't thing this is a behaviour intended by Traefik, but I may be wrong.

In Traefik a `service.name` tag is set on a span here: https://github.com/traefik/traefik/blob/702e0a461a3e49591772fac0c51cfe4bd4d3cfd8/pkg/middlewares/tracing/forwarder.go#L46
Unfortunatly, this tag is meaningful for Datadog, as we can see here in their tracing library used in Traefik: https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/ext/tags.go
The tag is interpreted as the service name of the span.

I contacted Datadog, which consider that it's not a issue, and the fix should be done on client (Traefik) side.

I tried last version of the dd-tracer-go library, with the same results.

A simple solution would be to set `serviceName` or `traefik.service.name` instead of `service.name`. 
To be consistent, `router.name` and `http.host` could also be renamed.

I can open a PR if you agree on this solution.

**Comment by mpl** [comment]
@julientard Hello. To clarify, it seems to me that in https://github.com/traefik/traefik/issues/7578 you were saying that what you're describing in #8062 is the root cause of the issue in #7578. Have you changed your mind on that, i.e. do you think they are distinct issues, or do you think that fixing #8062 also fixes #7578 ?

**Comment by julientard** [comment]
Hello @mpl, 
These two issues are different, but related. Fixing one could fix the other one, depending on the solution.

By checking again #7578, and my Datadog metrics, here is what's happening. I think @tazle speaks about [tracing applications metrics](https://docs.datadoghq.com/tracing/guide/metrics_namespace/) which are generated by Datadog from spans.
The naming comes directly from span metadata.

As you can see on my screenshot, in my case :
* the service name is `apps-cms-5fc20d3aa9c4dc543df3` (set by Traefik as described in the issue)
* the span name is `forward_apps_cms_5fc20d3aa9c4dc543df3_apps_cms_5fc20d3aa9c4dc543df3_kubernetescrd` (set [here](https://github.com/traefik/traefik/blob/702e0a461a3e49591772fac0c51cfe4bd4d3cfd8/pkg/middlewares/tracing/forwarder.go#L43)). => Metrics come from this span name.
![Screenshot from 2021-04-13 21-23-23](https://user-images.githubusercontent.com/16280863/114610068-cc860380-9c9f-11eb-98c1-36f37e5bff72.png)

So technically both are not related. One is the service name in which the span occurs, the other one is the span name. Contrary to what is being said in #7578, I find it acceptable to have different names in these two places.

In my opinion, to "fix" both issues:
* The service name should be left as default. It's something that the user want to set by himself (and you provide the option to do so, which is great). By doing this, the metric should disappear, because the span wouldn't be anymore the top level span. (the top level span would be ~EntryPoint_websecure_mydomain.com, as it should be).
* The span name could be left as it is currently. In our case, the most appropriate name would be the IngressRoute>metadata>name.

Thanks for your help, and for this great product !

**Comment by luckielordie** [comment]
I'm sorry, I'm experiencing the same behaviour as this issue and I'm eager to help solve it but I don't understand what you mean by your fix @julientard. 

Are there Traefik changes required? Or is it just configuration? I really don't want all of my IngressRoutes to appear as services, we have multiple domains pointing to services and it's excessive to visualise a domain as a service through DataDog.

**Comment by julientard** [comment]
Yes, a change in Traefik is required.
This [line](https://github.com/traefik/traefik/blob/702e0a461a3e49591772fac0c51cfe4bd4d3cfd8/pkg/middlewares/tracing/forwarder.go#L46) should be removed, or the key name should be different.

The "workaround" I found is to disable tracing on Traefik.

**Comment by luckielordie** [comment]
@julientard Feel free to add criticism to the PR. I hope you don't mind me taking it on myself to open a branch :)

**Comment by traefiker** [comment]
Closed by #8323.
